ExampleMod/Content/Items/Weapons/ExampleGun.cs

```diff
@@ -50,6 +52,18 @@ public class ExampleGun : ModItem
 			return new Vector2(2f, -2f);
 		}
 
+		// Every 3rd projectile this gun shoots will be an ExampleInstancedProjectile
+		public override bool Shoot(Player player, ProjectileSource_Item_WithAmmo source, Vector2 position, Vector2 velocity, int type, int damage, float knockback) {
+			if (Main.rand.NextBool(3))
+			{
+				Projectile.NewProjectile(source, position, velocity, ModContent.ProjectileType<ExampleInstancedProjectile>(), damage, knockback, player.whoAmI);
+
+				return false;
+			}
+			
+			return true;
+		}
```

Comment:
Since you are only changing the type of the projectile being shot with a chance, this is also achievable by changing `type` in `ModifyShootStats`, like so:
```suggestion
		public override void ModifyShootStats(Player player, ref Vector2 position, ref Vector2 velocity, ref int type, ref int damage, ref float knockback) {
			if (Main.rand.NextBool(3)) {
				type = ModContent.ProjectileType<ExampleInstancedProjectile>();
			}
		}
```